Resources/public/js/views/services/ez-locationviewviewservice.js

```diff
@@ -43,6 +44,136 @@ YUI.add('ez-locationviewviewservice', function (Y) {
         },
 
         /**
+         * sendToTrashAction event handler, sends location available in request to trash
+         * and makes the application navigate to parent location
+         *
+         * @method _sendContentToTrash
+         * @protected
+         * @param {Object} e event facade of the sendToTrashAction event
+         */
+        _sendContentToTrash: function (e) {
+            var app = this.get('app'),
+                that = this,
+                contentService = this.get('capi').getContentService(),
+                location = this.get('location'),
+                locationId = location.get('id'),
+                contentName = e.content.get('name');
+
+            e.preventDefault();
+            this._set('trackOutsideEvents', false);
+            this.fire('confirmBoxOpen', {
+                config: {
+                    title: "Are you sure you want to send this content to trash?",
+                    confirmHandler: Y.bind(function () {
+                        var tasks, parentLocation, trashPath, endLoadParent, endLoadTrashPath;
+
+                        that._sendContentToTrashNotificationStarted(locationId, contentName);
+
+                        tasks = new Y.Parallel();
+                        endLoadParent = tasks.add();
+                        that._loadParent(location, function (error, parentLocationResponse) {
+                            if (error) {
+                                that._sendContentToTrashNotificationError(locationId, contentName);
+                                that._error("Failed to load parent location of " + locationId);
+                                return;
+                            }
+
+                            parentLocation = parentLocationResponse.location;
+                            endLoadParent();
+                        });
+
+                        endLoadTrashPath = tasks.add();
+                        contentService.loadRoot(function (error, loadRootResponse) {
+                            if (error) {
+                                that._sendContentToTrashNotificationError(locationId, contentName);
+                                that._error("Failed to contact the REST API");
+                                return;
+                            }
+
+                            trashPath = loadRootResponse.document.Root.trash._href;
+                            endLoadTrashPath();
+                        });
+
+                        tasks.done(function () {
+                            contentService.moveSubtree(locationId, trashPath, function (err, response) {
```

Comment:
I think you should add a `trash` method to the location model and this method would discover the trash path, so that here, you would have only the parent loading and then the `trash` call.